tests/Functional/Parameters/BooleanFilterTest.php

```diff
@@ -0,0 +1,187 @@
+    {
+        $route = $this->getRoute($activeValue, $param);
+        $response = self::createClient()->request('GET', $route);
+        $this->assertResponseIsSuccessful();
+
+        $data = $response->toArray();
+        $entities = $data['hydra:member'];
+
+        $this->assertCount($expectedCount, $entities, \sprintf('The number of items with %s=%s should be %d', $param, $activeValue, $expectedCount));
+
+        if ('null' !== $activeValue) {
+            foreach ($entities as $entity) {
+                $isActiveExpected = \in_array($activeValue, ['true', '1'], true);
+                $expectedValue = $isActiveExpected ? 'true' : 'false';
+                $message = \sprintf("Expected 'active' to be %s", $expectedValue);
+
+                $this->assertSame($isActiveExpected, $entity['active'], $message);
+            }
+        }
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     */
+    private function assertValidationErrorResponse(string $activeValue, string $param = 'active'): void
+    {
+        $route = $this->getRoute($activeValue, $param);
+        $response = self::createClient()->request('GET', $route);
+
+        $this->assertEquals(422, $response->getStatusCode(), 'Expected status code 422 for validation error.');
+        $this->assertJsonContains([
+            '@context' => '/contexts/ConstraintViolationList',
+            '@id' => '/validation_errors/c1051bb4-d103-4f74-8988-acbcafc7fdc3',
+            '@type' => 'ConstraintViolationList',
+            'status' => 422,
+            'violations' => [
+                0 => [
+                    'propertyPath' => 'active',
+                    'message' => 'This value should not be blank.',
+                    'code' => 'c1051bb4-d103-4f74-8988-acbcafc7fdc3',
+                ],
+            ],
+            'detail' => 'active: This value should not be blank.',
+            'description' => 'active: This value should not be blank.',
+            'type' => '/validation_errors/c1051bb4-d103-4f74-8988-acbcafc7fdc3',
+            'title' => 'An error occurred',
+            'hydra:description' => 'active: This value should not be blank.',
+            'hydra:title' => 'An error occurred',
+        ]);
+    }
+
+    private function loadFixtures(string $resource): void
+    {
+        $manager = $this->getManager();
+
+        $entitiesData = [true, true, false, null];
+        foreach ($entitiesData as $activeValue) {
+            $entity = new $resource(active: $activeValue);
+            $manager->persist($entity);
+        }
+
+        $manager->flush();
+    }
+
+    private function getRoute(?string $value, string $param = 'active'): string
+    {
+        return \sprintf('%s?%s=%s', self::ROUTE, $param, $value);
```

Comment:
not a fan of this as well, it makes tests too abstract, I'd like to know when reading a test how the route is, looking for the logic inside the test case is not convenient imo